pkg/provider/beyondtrust/provider_test.go

```diff
@@ -0,0 +1,226 @@
+
+		default:
+			http.NotFound(w, r)
+		}
+	}))
+	t.Cleanup(server.Close)
+
+	clientConfig := clientcmd.NewDefaultClientConfig(clientcmdapi.Config{
+		Clusters: map[string]*clientcmdapi.Cluster{
+			"test": {
+				Server: server.URL,
+			},
+		},
+		AuthInfos: map[string]*clientcmdapi.AuthInfo{
+			"test": {
+				Token: "token",
+			},
+		},
+		Contexts: map[string]*clientcmdapi.Context{
+			"test": {
+				Cluster:  "test",
+				AuthInfo: "test",
+			},
+		},
+		CurrentContext: "test",
+	}, &clientcmd.ConfigOverrides{})
+
+	restConfig, err := clientConfig.ClientConfig()
+	assert.Nil(t, err)
+	c, err := kubeclient.New(restConfig, kubeclient.Options{})
+	assert.Nil(t, err)
+
+	return c
+}
+
+func TestNewClient(t *testing.T) {
+	t.Run("should create new client", func(t *testing.T) {
+		ctx := context.Background()
+		p := &Provider{}
+		c := createMockPasswordSafeClient(t)
+		s := esv1beta1.SecretStore{
+			Spec: esv1beta1.SecretStoreSpec{
+				Provider: &esv1beta1.SecretStoreProvider{
+					Beyondtrust: &esv1beta1.BeyondtrustProvider{
+						APIURL:        "https://test.ps-dev.beyondtrustcloud.com:443/BeyondTrust/api/public/v3/",
+						Retrievaltype: "SECRET",
+						Clientid: &esv1beta1.BeyondTrustProviderSecretRef{
+							Value: "12345678-25fg-4b05-9ced-35e7dd5093ae",
+						},
+						Clientsecret: &esv1beta1.BeyondTrustProviderSecretRef{
+							Value: "1234567890on8mTcOzJcltiEg4wYOhDVMer123456789",
+						},
+					},
+				},
+			},
+		}
+
+		_, err := p.NewClient(ctx, &s, c, "test")
+
+		assert.Nil(t, err)
+	})
+
+}
+
+func TestNewClientBadClientId(t *testing.T) {
+	t.Run("should create new client", func(t *testing.T) {
+		ctx := context.Background()
+		p := &Provider{}
+		c := createMockPasswordSafeClient(t)
+		s := esv1beta1.SecretStore{
+			Spec: esv1beta1.SecretStoreSpec{
+				Provider: &esv1beta1.SecretStoreProvider{
+					Beyondtrust: &esv1beta1.BeyondtrustProvider{
+						APIURL:        "https://test.ps-dev.beyondtrustcloud.com:443/BeyondTrust/api/public/v3/",
+						Retrievaltype: "SECRET",
+						Clientid: &esv1beta1.BeyondTrustProviderSecretRef{
+							Value: "6138d050",
+						},
+						Clientsecret: &esv1beta1.BeyondTrustProviderSecretRef{
+							Value: "1234567890on8mTcOzJcltiEg4wYOhDVMer123456789",
+						},
+					},
+				},
+			},
+		}
+
+		_, err := p.NewClient(ctx, &s, c, "test")
+
+		assert.Equal(t, err.Error(), "error: Key: 'UserInputValidaton.ClientId' Error:Field validation for 'ClientId' failed on the 'min' tag", "The two errors should be the same.")
+	})
+
+}
+
+func TestNewClientBadClientSecret(t *testing.T) {
+	t.Run("should create new client", func(t *testing.T) {
+		ctx := context.Background()
+		p := &Provider{}
+		c := createMockPasswordSafeClient(t)
+		s := esv1beta1.SecretStore{
+			Spec: esv1beta1.SecretStoreSpec{
+				Provider: &esv1beta1.SecretStoreProvider{
+					Beyondtrust: &esv1beta1.BeyondtrustProvider{
+						APIURL:        "https://test.ps-dev.beyondtrustcloud.com:443/BeyondTrust/api/public/v3/",
```

Comment:
please change to APIURL: "[https://example.com:443/BeyondTrust/api/public/v3/](https://example.com/BeyondTrust/api/public/v3/)",